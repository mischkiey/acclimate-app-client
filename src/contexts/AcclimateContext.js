import React from 'react';

const AcclimateContext = React.createContext({});

export default AcclimateContext;