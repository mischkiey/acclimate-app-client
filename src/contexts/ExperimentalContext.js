import React, { Component } from 'react';
import TokenService from '../services/token-services';

export const ExperimentalContext = React.createContext({});

export class ExperimentalProvider extends Component {
    state = {
        log: false,
        tasks: [],
        shoppingItems: [],
    };

    handleUserLog = () => {
        const log = TokenService.hasAuthToken();
        this.setState({log});
    };

    handleUserLogOut = () => {
        TokenService.clearAuthToken();
        this.handleUserLog();
    };

    setTasks = (tasks) => {
        this.setState({tasks})
    };

    setShoppingItems = (shoppingItems) => {
        this.setState({shoppingItems})
    };

    render() {
        const value = {
            log: this.state.log,
            tasks: this.state.tasks,
            shoppingItems: this.state.shoppingItems,

            handleUserLog: this.handleUserLog,
            handleUserLogOut: this.handleUserLogOut,
            setTasks: this.setTasks,
            setShoppingItems: this.setShoppingItems,
        };

        return (
            <ExperimentalContext.Provider value={value}>
                {this.props.children}
            </ExperimentalContext.Provider>
        );
    };
};

