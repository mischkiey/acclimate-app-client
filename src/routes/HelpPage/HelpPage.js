import React, { Component } from 'react';

class HelpPage extends Component {
    render() {
        return (
            <div className='white'>
                <i className="material-icons">face</i>
                <h2>Description:</h2>
                <p>
                    Acclimate is an informational and utility disaster management app that aims to provide accurate, reliable, and useful information and tools to be able to handle impending disasters systematically and effectively
                </p>
                
                <h2>How It Works:</h2>
                <ol>
                    <li>
                        The app provides users with general information on a selected disaster
                    </li>
                    <li>
                        The app will generate a disaster management plan (based on user-selection) that has steps for before, during, and after the event
                    </li>
                    <li>
                        The app will also generate a task list that incorporates the above plan
                    </li>
                    <li>
                        The app will also generate a shopping list with recommended supplies
                    </li>
                    <li>
                        Users can personalize the task and shopping lists to tailorfit their current circumstances/needs
                    </li>
                </ol>

                <h2>Credits</h2>
            </div>
        );
    };
};

export default HelpPage;