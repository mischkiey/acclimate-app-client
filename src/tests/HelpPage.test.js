import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { AcclimateProvider } from '../contexts/AcclimateContext';
import HelpPage from '../routes/HelpPage/HelpPage';

describe(`HelpPage Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <AcclimateProvider>
                        <HelpPage />
                    </AcclimateProvider>
                </BrowserRouter>,
                div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <AcclimateProvider>
                            <HelpPage />
                        </AcclimateProvider>
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})