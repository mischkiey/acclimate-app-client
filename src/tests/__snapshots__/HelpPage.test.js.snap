// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpPage Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="white"
>
  <i
    className="material-icons"
  >
    help_outline
  </i>
  <h2>
    Hello and welcome to Acclimate!
  </h2>
  <p>
    Acclimate is an informational and utility disaster management app that aims to provide accurate, reliable, and useful information and tools to be able to handle impending disasters systematically and effectively
  </p>
  <hr />
  <h2>
    How It Works:
  </h2>
  <ol>
    <li>
      The app provides the user with general information on a selected disaster
    </li>
    <li>
      The app provides disaster management plans that has steps for before, during, and after the event
    </li>
    <li>
      The app provides the user with a highly-customizable task list and shopping list
    </li>
    <li>
      Users may add steps from the chosen disaster plan to their lists where appropriate at the push of a button
    </li>
  </ol>
  <hr />
  <h2>
    Hints and Tips:
  </h2>
  <ul>
    <li>
      <i
        className="material-icons"
      >
        help_outline
      </i>
       - Displays help page
    </li>
    <li>
      <i
        className="material-icons"
      >
        search
      </i>
       - Search database for a disaster management plan
    </li>
    <li>
      <i
        className="material-icons"
      >
        dashboard
      </i>
       - View your dashboard
    </li>
    <li>
      <i
        className="material-icons"
      >
        power_settings_new
      </i>
       - Log out securely
    </li>
    <li>
      <i
        className="material-icons"
      >
        add_circle
      </i>
       - Add item
    </li>
    <li>
      <i
        className="material-icons"
      >
        add_task
      </i>
       - Add disaster plan step to task list (automatically creates a shorthand version)
    </li>
    <li>
      <i
        className="material-icons"
      >
        add_shopping_cart
      </i>
       - Add disaster plan step to shopping list (automatically creates a shorthand version)
    </li>
    <li>
      <i
        className="material-icons"
      >
        delete
      </i>
       - Delete item
    </li>
  </ul>
  <h2>
    Acknowledgments:
  </h2>
  <ul>
    <li>
      Al Zimmerman - TA
    </li>
    <li>
      Alfredo Salazar - Mentor
    </li>
    <li>
      Joshua Struve - Overall design
    </li>
  </ul>
  <h2>
    Credits and References
  </h2>
  <ul>
    <li>
      FEMA - https://www.fema.gov/
    </li>
    <li>
      Red Cross - https://www.redcross.org/
    </li>
    <li>
      Unsplash - https://unsplash.com/
    </li>
  </ul>
</div>
`;
